package/MDAnalysis/analysis/density.py

```diff
@@ -511,7 +512,13 @@ def current_coordinates():
     for ts in u.trajectory:
         print("Histograming %6d atoms in frame %5d/%d  [%5.1f%%]\r" % \
               (len(coord), ts.frame, u.trajectory.n_frames, 100.0 * ts.frame / u.trajectory.n_frames),)
-        coord = current_coordinates()
+        if sel_update=True:
+           group=u.select_atoms(atomselection)
+           coord=group.coordinates()
+           
+        else:
+           coord = current_coordinates()
```

Comment:
use `positions` instead of `coordinates()` (`coordinates()` is not wrong but we prefer `positions`)